spring-amqp/src/main/java/org/springframework/amqp/support/converter/SerializerMessageConverter.java

```diff
@@ -120,11 +141,40 @@ public Object fromMessage(Message message) throws MessageConversionException {
 		return content;
 	}
 
+	private Object deserialize(ByteArrayInputStream inputStream, Deserializer<Object> deserializer)
+			throws IOException {
+		ClassLoader classLoader = null;
+		try {
+			classLoader = (ClassLoader) new DirectFieldAccessor(deserializer).getPropertyValue("classLoader");
```

Comment:
After some private discussion we decided to leave the solution as is without any general extraction and freeing of the injection.

Will be merged after polishing.